File: SozlukJava/app/src/main/java/com/example/sozlukjava/adapter/EntryAdapter.java
package com.example.sozlukjava.adapter;

// EntryAdapter.java
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.sozlukjava.R;
import com.example.sozlukjava.model.Entry;

import java.util.List;

public class EntryAdapter extends RecyclerView.Adapter<EntryAdapter.EntryViewHolder> {

    public interface OnEntryClick {
        void onClick(Entry entry);
    }

    public interface OnEntryLongClick {
        void onLongClick(Entry entry);
    }

    private final List<Entry> entryList;
    private final OnEntryClick onClick;
    private final OnEntryLongClick onLongClick;

    public EntryAdapter(List<Entry> entryList, OnEntryClick onClick, OnEntryLongClick onLongClick) {
        this.entryList = entryList;
        this.onClick = onClick;
        this.onLongClick = onLongClick;
    }

    static class EntryViewHolder extends RecyclerView.ViewHolder {
        TextView tvBaslik, tvIcerik, tvUsername, tvTarih;

        EntryViewHolder(@NonNull View itemView) {
            super(itemView);
            tvBaslik = itemView.findViewById(R.id.tvBaslik);
            tvIcerik = itemView.findViewById(R.id.tvIcerik);
            tvUsername = itemView.findViewById(R.id.tvUsername);
            tvTarih = itemView.findViewById(R.id.tvTarih);
        }
    }

    @NonNull
    @Override
    public EntryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.entry_card, parent, false);
        return new EntryViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull EntryViewHolder holder, int position) {
        Entry entry = entryList.get(position);
        holder.tvBaslik.setText(entry.getTitle());
        holder.tvIcerik.setText(entry.getContent());
        holder.tvUsername.setText(entry.getUsername());

        String created = entry.getCreated_at();
        if (created != null && created.length() >= 10) {
            holder.tvTarih.setText(created.substring(0, 10));
        } else {
            holder.tvTarih.setText("");
        }

        holder.itemView.setOnClickListener(v -> {
            if (onClick != null) onClick.onClick(entry);
        });
        holder.itemView.setOnLongClickListener(v -> {
            if (onLongClick != null) onLongClick.onLongClick(entry);
            return true;
        });
    }

    @Override
    public int getItemCount() {
        return entryList != null ? entryList.size() : 0;
    }
}

File: SozlukKotlin/app/src/main/java/com/example/sozlukkotlin/adapter/EntryAdapter.kt
package com.example.sozlukkotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.sozlukkotlin.R
import com.example.sozlukkotlin.model.Entry

class EntryAdapter(
    private val entryList: List<Entry>,
    private val onClick: (Entry) -> Unit,
    private val onLongClick: (Entry) -> Unit
) : RecyclerView.Adapter<EntryAdapter.EntryViewHolder>() {

    class EntryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvBaslik: TextView = itemView.findViewById(R.id.tvBaslik)
        val tvIcerik: TextView = itemView.findViewById(R.id.tvIcerik)
        val tvUsername: TextView = itemView.findViewById(R.id.tvUsername)
        val tvTarih: TextView = itemView.findViewById(R.id.tvTarih)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EntryViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.entry_card, parent, false)
        return EntryViewHolder(view)
    }

    override fun getItemCount(): Int = entryList.size

    override fun onBindViewHolder(holder: EntryViewHolder, position: Int) {
        val entry = entryList[position]
        holder.tvBaslik.text = entry.title
        holder.tvIcerik.text = entry.content
        holder.tvUsername.text = entry.username

        // Tarihi formatla (örnek: "2025-08-07 13:45:00")
        holder.tvTarih.text = entry.created_at.substring(0, 10)

        holder.itemView.setOnClickListener {
            onClick(entry)
        }
        holder.itemView.setOnLongClickListener {
            onLongClick(entry)
            true
        }
    }

}
